WORKSPACE

```diff
@@ -83,6 +83,20 @@ go_repository(
     importpath = "github.com/golang/mock",
 )
 
+#Get the solidity compiler
+
+load("//contracts:rules/solcompiler.bzl", "solc_fetch")
+
+solc_fetch(
+    name = "solc_sdk",
+    sdks = {
+        "linux_amd64": (
```

Comment:
Can you add all available OS?

Reply:
Most mac users use homebrew to install solc https://github.com/ethereum/go-ethereum/wiki/Installation-Instructions-for-Mac
Not sure if that's something we can do here

Reply:
Would the same binary for linux work on mac? How do they build for mac? What is provided by homebrew?

Reply:
To answer your question @hubris, I prefer the first of your three suggestions. 

If we can't find compiled binaries online, let's build it and check it into the repo. I'm fine with not supporting 32 bit systems for now, but we need to support darwin and windows at a minimum.

Reply:
hey @hubris any update on this?

Reply:
@rauljordan Hi, I will work on this tomorrow.

Reply:
@prestonvanloon Hi, I tried to add the win64 compiler but I'm getting in troubles just for building prysm on windows.
I downloaded bazel for windows, cloned the repo and from the master, I did ``bazel test //...``. I got:
```bash
INFO: Build options have changed, discarding analysis cache.
ERROR: D:/devel/gopath/src/github.com/hubris/prysm/client/BUILD.bazel:21:1: every rule of type app_layer implicitly depends upon the target '@gzip//:gzip', but this target could not be found because of: no such package '@gzip//': expected value of type 'string or Label or path' for parameter 'from', in method call symlink(NoneType, string) of 'repository_ctx'
ERROR: Analysis of target '//client:image' failed; build aborted: no such package '@gzip//': expected value of type 'string or Label or path' for parameter 'from', in method call symlink(NoneType, string) of 'repository_ctx'
INFO: Elapsed time: 2.209s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (47 packages loaded)
FAILED: Build did NOT complete successfully (47 packages loaded)
````

Did I miss something?

Reply:
@hubris I am not sure that the docker image will build on windows.

You should be able to build the specific target binary. `bazel build //beacon-chain`
Are you synced with master? I thought we add the manual tag to that target so it wouldn't build when you build `//...`.

Reply:
@prestonvanloon Thank you for the answer. I tried with latest master (42d93812ced4c5a219371f141fe80ec07a658c52).
Doing ``bazel build //beacon-chain`` in cmd.exe still give me errors (doing the same in bash gives similar errors):
```
DEBUG: C:/users/hubris/_bazel_hubris/gcsjzdpj/external/bazel_tools/tools/cpp/lib_cc_configure.bzl:115:5:
Auto-Configuration Warning: 'BAZEL_VC' is not set, start looking for the latest Visual C++ installed.
DEBUG: C:/users/hubris/_bazel_hubris/gcsjzdpj/external/bazel_tools/tools/cpp/lib_cc_configure.bzl:115:5:
Auto-Configuration Warning: Looking for VS%VERSION%COMNTOOLS environment variables, eg. VS140COMNTOOLS
DEBUG: C:/users/hubris/_bazel_hubris/gcsjzdpj/external/bazel_tools/tools/cpp/lib_cc_configure.bzl:115:5:
Auto-Configuration Warning: Visual C++ build tools found at D:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\
INFO: SHA256 (https://codeload.github.com/google/protobuf/zip/106ffc04be1abf3ff3399f54ccf149815b287dd9) = 4d127e1c3608803bc845cc4b3c8c42a6a16263b6ddcb932b53049e7ec8101398
ERROR: D:/devel/gopath/src/github.com/prysmaticlabs/prysm/beacon-chain/BUILD.bazel:5:1: no such package '@com_github_urfave_cli//': no such package '@bazel_gazelle_go_repository_tools//': no such package '@com_github_pelletier_go_toml//': Traceback (most recent call last):
        File "C:/users/hubris/_bazel_hubris/gcsjzdpj/external/io_bazel_rules_go/go/private/tools/overlay_repository.bzl", line 67
                _apply_overlay(ctx, overlay)
        File "C:/users/hubris/_bazel_hubris/gcsjzdpj/external/io_bazel_rules_go/go/private/tools/overlay_repository.bzl", line 95, in _apply_overlay
                _check_execute(ctx, ["cp", src_path, dst_rel], ("fa...))
        File "C:/users/hubris/_bazel_hubris/gcsjzdpj/external/io_bazel_rules_go/go/private/tools/overlay_repository.bzl", line 100, in _check_execute
                fail((((message + "\n") + res.stdout)...))
failed to copy file from C:/users/hubris/_bazel_hubris/gcsjzdpj/external/io_bazel_rules_go/third_party/com_github_pelletier_go_toml/cmd/tomljson/BUILD.bazel.in
java.io.IOException: ERROR: src/main/native/windows/processes-jni.cc(239): CreateProcessW("cp" C:/users/hubris/_bazel_hubris/gcsjzdpj/external/io_bazel_rules_go/third_party/com_github_pelletier_go_toml/cmd/tomljson/BUILD.bazel.in cmd/tomljson/BUILD.bazel): The system cannot find the file specified.
 and referenced by '//beacon-chain:go_default_library'
ERROR: Analysis of target '//beacon-chain:beacon-chain' failed; build aborted: no such package '@com_github_urfave_cli//': no such package '@bazel_gazelle_go_repository_tools//': no such package '@com_github_pelletier_go_toml//': Traceback (most recent call last):
        File "C:/users/hubris/_bazel_hubris/gcsjzdpj/external/io_bazel_rules_go/go/private/tools/overlay_repository.bzl", line 67
                _apply_overlay(ctx, overlay)
        File "C:/users/hubris/_bazel_hubris/gcsjzdpj/external/io_bazel_rules_go/go/private/tools/overlay_repository.bzl", line 95, in _apply_overlay
                _check_execute(ctx, ["cp", src_path, dst_rel], ("fa...))
        File "C:/users/hubris/_bazel_hubris/gcsjzdpj/external/io_bazel_rules_go/go/private/tools/overlay_repository.bzl", line 100, in _check_execute
                fail((((message + "\n") + res.stdout)...))
failed to copy file from C:/users/hubris/_bazel_hubris/gcsjzdpj/external/io_bazel_rules_go/third_party/com_github_pelletier_go_toml/cmd/tomljson/BUILD.bazel.in
java.io.IOException: ERROR: src/main/native/windows/processes-jni.cc(239): CreateProcessW("cp" C:/users/hubris/_bazel_hubris/gcsjzdpj/external/io_bazel_rules_go/third_party/com_github_pelletier_go_toml/cmd/tomljson/BUILD.bazel.in cmd/tomljson/BUILD.bazel): The system cannot find the file specified.
INFO: Elapsed time: 4.093s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (7 packages loaded)
```

Reply:
These errors are weird:
``no such package '@com_github_urfave_cli//': no such package '@bazel_gazelle_go_repository_tools//': no such package '@com_github_pelletier_go_toml//': ``

Reply:
OK, maybe we don't have build on windows support. I believe no one is working on windows so this might have been broken for some time. 

Let's go ahead with win64 as you have it and assume that it will work when the other issues are resolved.

Where you able to find or build a binary for mac?

Reply:
@prestonvanloon I didn't find a binary for mac and I don't have one to build. If you can provide me a binary I could try to integrate it.

Reply:
The homebrew uses the tar.gz. 
See: https://github.com/ethereum/homebrew-ethereum/blob/master/solidity.rb

Reply:
Although it uses cmake to build it. Maybe we need to submit a feature request to solidity to start providing the mac binaries.

Reply:
@prestonvanloon  There is one already: https://github.com/ethereum/solidity/issues/3168